src/overlays/actors/ovl_Obj_Dora/z_obj_dora.c

```diff
@@ -98,34 +152,188 @@ static DamageTable D_80B61488 = {
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderTrisElements); i++) {
+        for (j = 0; j < ARRAY_COUNT(vtx); j++) {
+            Matrix_MultiplyVector3fByState(&sTrisElementsInit[i].dim.vtx[j], &vtx[j]);
+        }
+        Collider_SetTrisVertices(&this->colliderTris, i, &vtx[0], &vtx[1], &vtx[2]);
+    }
+    ObjDora_SetupWait(this);
+}
+
+void ObjDora_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDora* this = THIS;
+
+    Collider_DestroyTris(globalCtx, &this->colliderTris);
+}
+
+void ObjDora_SetupWait(ObjDora* this) {
+    this->actionFunc = ObjDora_Wait;
+}
+
+void ObjDora_Wait(ObjDora* this, GlobalContext* globalCtx) {
+}
+
+void ObjDora_SetupMoveGong(ObjDora* this) {
+    if (this->lastGongHitType == DORA_HIT_LIGHT) {
+        this->gongForce.x = 2.0f;
+        this->gongForce.y = 5461.0f;
+        this->gongForce.z = 1820.0f;
+    } else {
+        this->gongForce.x = 4.0f;
+        this->gongForce.y = 12743.0f;
+        this->gongForce.z = 5461.0f;
+    }
+    this->gongAngle.x = 0;
+    this->gongAngle.z = 0;
+    this->actionFunc = ObjDora_MoveGong;
+}
+
+void ObjDora_MoveGong(ObjDora* this, GlobalContext* globalCtx) {
+
+    if ((this->gongForce.y < 182.0f) && (this->gongForce.z < 182.0f)) {
+        this->lastGongHitType = DORA_HIT_LIGHT;
+        ObjDora_SetupWait(this);
+    }
+
+    Math_SmoothStepToF(&(this->gongForce.x), 0, 0.2f, 0.2f, 0.1f);
+    Math_SmoothStepToF(&(this->gongForce.y), 0, 0.5f, 54.0f, 18.0f);
+    Math_SmoothStepToF(&(this->gongForce.z), 0, 0.5f, 54.0f, 18.0f);
+    this->gongAngle.x += 0x1555;
+    this->gongAngle.z += 0x238E;
+    this->gongRotation.x = Math_SinS(this->gongAngle.x) * this->gongForce.y;
+    this->gongRotation.z = Math_SinS(this->gongAngle.z) * this->gongForce.z;
+}
+
+bool ObjDora_IsHalfHourly(u16 time) {
+    f32 temp = time;
+    temp -= (1365.3334f * (s32)(time / 1365.3334f));
+
+    if ((temp < 45.511112f) || (1319.8223f < temp)) {
+        return true;
+    }
+
+    return false;
+}
+
+void ObjDora_CheckCollision(ObjDora* this, GlobalContext* globalCtx) {
+    Actor* itemDrop;
+    u16 time;
+
+    if (this->colliderTris.base.acFlags & AC_HIT) {
+        time = gSaveContext.save.time;
+        this->colliderTris.base.acFlags &= ~AC_HIT;
+        this->collisionCooldownTimer = 5;
+
+        switch (this->actor.colChkInfo.damageEffect) {
+            case 14:
+            case 15:
+                if (this->actor.colChkInfo.damageEffect == 15) {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_DORA_S);
+                    this->lastGongHitType = DORA_HIT_LIGHT;
+                } else {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_DORA_L);
+                    this->lastGongHitType = DORA_HIT_STRONG;
+                }
+
+                func_800BC848(&this->actor, globalCtx, 5, 10);
+                ObjDora_SetupMoveGong(this);
+
+                if ((ObjDora_IsHalfHourly(time) == 1) && (this->rupeeDropTimer == 0)) {
```

Comment:
```suggestion
                if (ObjDora_IsHalfHourly(time) && (this->rupeeDropTimer == 0)) {
```
Removing the `== 1` (which is `== true`) is generally preferred by us when this is identified as a true/false return for the function.

Reply:
nvm, checked and it doesn't match here, ignore this